services/web/server/src/simcore_service_webserver/catalog/_api.py

```diff
@@ -56,9 +60,113 @@ def create(cls, request: Request) -> "CatalogRequestContext":
             )
 
 
+async def _safe_replace_service_input_outputs(
+    service: dict[str, Any], unit_registry: UnitRegistry
+):
+    try:
+        await asyncio.to_thread(
+            replace_service_input_outputs,
+            service,
+            unit_registry=unit_registry,
+            **RESPONSE_MODEL_POLICY,
+        )
+    except KeyError:  # noqa: PERF203
+        # This will limit the effect of a any error in the formatting of
+        # service metadata (mostly in label annotations). Otherwise it would
+        # completely break all the listing operation. At this moment,
+        # a limitation on schema's $ref produced an error that made faiing
+        # the full service listing.
+        _logger.exception(
+            "Failed while processing this %s. "
+            "Skipping service from listing. "
+            "TIP: check formatting of docker label annotations for inputs/outputs.",
```

Comment:
Q: is this one of the issues or could it be multiple reasons why this breaks? Let's not have possibly misleading tips

Reply:
This is a safe-guard from the old code. 
The problem is that we have no guarantees on the labels injected in the images. The registry is for the moment not validated. I will definitively follow up on this one once we get this first refactoring done